public/app/features/alerting/unified/utils/rules.ts

```diff
@@ -277,7 +277,13 @@ export const flattenCombinedRules = (rules: CombinedRuleNamespace[]) => {
     groups.forEach(({ name: groupName, rules }) => {
       rules.forEach((rule) => {
         if (rule.promRule && isAlertingRule(rule.promRule)) {
-          acc.push({ dataSourceName: getRulesSourceName(rulesSource), namespaceName, groupName, ...rule });
+          acc.push({
+            ...rule,
+            dataSourceName: getRulesSourceName(rulesSource),
+            namespaceName,
+            groupName,
+            namespace: { ...rule.namespace, uid: rule.promRule.folderUid },
```

Comment:
Does it work for vanilla prom rules?

Reply:
The panel still works with this for other rules, and we only supported filtering by folder UID for GMA - so it shouldn't matter here